### PR TITLE
Essential fix #1 (visibility unit showing undefined)

### LINKED ISSUE #1: Darksky forecast date format
The Darksky forecast date format is currently US i.e. M/D Could there please be an option to change that to D/M for those of us down in the colonies using dd/mm/yy   :)

### PR DESCRIPTION
**Pull request by ArendPWS**
It is really just this fix, nothing else. I really have no idea how I would be able to clarify any further.

### CODE REVIEW

**Comment by poblabs** [comment]
This PR shows over 145 added lines of modified code, and 121 lines removed. Is the visibility fix really this many lines?

This is showing a lot of websocket changes, graph changes, etc. 

When you commit your changes, you only need to commit the few lines that are the change and not everything. Otherwise things will become out of sync. This is why I say there's conflicts and I cannot merge. 

Each change should be it's own PR and not bundled. That way it's easy to track and rollback if needed in future.

File: skins/Belchertown/index.html.tmpl [hunk 1]
```diff
@@ -20,137 +20,147 @@
         
         var homepage_graphgroup = "$Extras.highcharts_homepage_graphgroup";
         
-        ajaxweewx(); // Initial call to load from weewx (date, daily high, low, etc)
-        
-        #if $Extras.has_key("forecast_enabled") and $Extras.forecast_enabled == '1'
-        // Load forecast
-        ajaxforecast(); // Initial call to load forecast data like 8 day outlook, weather icon and observation text
-        #end if
+        ajaxweewx().then(function(weewx_data) { // This call will make sure json/weewx_data.json is loaded before anything else
```
Comment (poblabs) [inline skins/Belchertown/index.html.tmpl:23]: This should be it's own PR?

File: skins/Belchertown/index.html.tmpl [hunk 2]
```diff
@@ -20,137 +20,147 @@
...
-            }
-            #end if
+                #if $Extras.has_key('earthquake_enabled') and $Extras.earthquake_enabled == '1'            
+                if ( moment.unix( "$earthquake_time" ).isValid() ) {
+                    // Check that the time is the time (number) and not the "no earthquake available" wording
+                    jQuery(".earthquake-time").html( moment.unix( "$earthquake_time" ).utcOffset($moment_js_utc_offset).format( "$obs.label.time_earthquake" ) );
+                } else {
+                    jQuery(".earthquake-time").html( "$earthquake_time" );
+                }
+                #end if
```
Comment (poblabs) [inline skins/Belchertown/index.html.tmpl:61]: Is all this needed in this PR? Looks like a lot of redundant changes?

File: skins/Belchertown/index.html.tmpl [hunk 3]
```diff
@@ -20,137 +20,147 @@
...
+                } else {
+                    sessionStorage.setItem ( "forecastInterval" , null );
+                    forecast_select( sessionStorage.getItem( "defaultInterval") );
+                }
+            });
+        }).catch(function(e) {
+            console.log(e);
         });
         
     </script>
```
Comment (poblabs) [inline skins/Belchertown/index.html.tmpl:166]: I'm confused why this is all within this PR? I think this code is fine? Where is the change?

File: skins/Belchertown/js/belchertown.js.tmpl [hunk 4]
```diff
@@ -102,7 +102,12 @@ jQuery(document).ready(function() {
...
 #if $Extras.has_key("theme") and $Extras.theme == 'auto'
 // Run this on every page for dark mode if skin theme is auto
-ajaxweewx();
+ajaxweewx().then(function(weewx_data) { // This call will make sure json/weewx_data.json is loaded before anything else
+	update_weewx_data(weewx_data); // Initial call to update (date, daily high, low, etc)
+	belchertown_debug(weewx_data); // Make weewx_data.json available in debugging console
+}).catch(function(e) {
+	console.log(e);
+});
 #end if
```
Comment (poblabs) [inline skins/Belchertown/js/belchertown.js.tmpl:111]: Should this be it's own PR?

File: skins/Belchertown/js/belchertown.js.tmpl [hunk 5]
```diff
@@ -581,10 +586,14 @@ function changeTheme(themeName, toggleOverride=false) {
...
-    jQuery.getJSON(get_relative_url() + "/json/weewx_data.json", update_weewx_data);
-};
+async function ajaxweewx() {
+    resp = await fetch(get_relative_url() + "/json/weewx_data.json");
+    if (!resp.ok) {
+        throw new Error("HTTP error! Unable to load weewx_data.json");
+    } else {	
+        return await resp.json();
+    }
+}
```
Comment (poblabs) [inline skins/Belchertown/js/belchertown.js.tmpl:597]: Should this be it's own PR?

File: skins/Belchertown/js/belchertown.js.tmpl [hunk 6]
```diff
@@ -1680,11 +1689,16 @@ function update_current_wx(data) {
...
+        ajaxweewx().then(function(weewx_data) { // This call will make sure json/weewx_data.json is loaded before anything else
+            setTimeout(update_weewx_data.bind(null, weewx_data), 10000); // Initial call to update (date, daily high, low, etc)
+            setTimeout(belchertown_debug.bind(null, weewx_data), 10000); // Make weewx_data.json available in debugging console
+            #if $Extras.has_key("forecast_enabled") and $Extras.forecast_enabled == '1'
+            setTimeout( ajaxforecast, 10000 ); // Update forecast data
+            #end if
+            setTimeout( ajaximages, 10000 ); // Update radar and home page hook "img src" if present
+        }).catch(function(e) {
+            console.log(e);
+        });
```
Comment (poblabs) [inline skins/Belchertown/js/belchertown.js.tmpl:1701]: Should this be it's own PR?

**Review by poblabs — COMMENTED** [review]
With the amount of changes here, it's hard for me to see the *actual* change that would fix the issue. That's what my "conflict" is based off of.

**Comment by ArendPWS** [comment]
I have tried to explain that this is caused by the old code being included in the new code, that gives this weird looking result in the Github compare. There are multiple calls to ajaxweewx() which all need to modified. It is not logical to create a PR for each individual modified call.

**Comment by poblabs** [comment]
> I have tried to explain that this is caused by the old code being included in the new code, that gives this weird looking result in the Github compare. There are multiple calls to ajaxweewx() which all need to modified. It is not logical to create a PR for each individual modified call.

I know why this is happening - you can include only the lined portions to submit to the PR - you don't have to include **everything**. That's what I've tried to explain 😄

**Comment by poblabs** [comment]
My suggestion - again - is to include only the lines that are changed and do not include the old code. 

Why?

Because your PR would then re-introduce old code into the master brannch effectively undo-ing a lot of work.

**Comment by ArendPWS** [comment]
No I mean it this way:

New code {
   Old code
}

That is the cause of all those apparent changes.

**Comment by ArendPWS** [comment]
There is no old code in this PR

**Comment by poblabs** [comment]
This section looks identical to me - why is it included? 

https://github.com/poblabs/weewx-belchertown/pull/533/files#diff-bbaea3b6cf0fd48299f330940404a9a8ff68a245903679c1c1e33a2b884cd964R31-R37

**Comment by poblabs** [comment]
Ah, it's indentation from the `ajaxweewx().then(function(weewx_data) {` section?

**Comment by ArendPWS** [comment]
YES!!!! No we are getting on the same frequency 👍 Maybenow  you have some idea why the compare looks the way it does :smiley:
